src/hello-world.cpp

```diff
@@ -0,0 +1,10 @@
+#include <cstdlib>
+#include <iostream>
+#include <string>
+
+std::string say_hello() { return std::string("Hello, CMake superbuild!"); }
+
+int main() {
```

Comment:
So two things. First, in C++ it's customary to put the definition of the main function in a file `main.cpp`. Second your CMakeLists.txt isn't building an executable (it's building a library) so there shouldn't be a main function (all hell's going to break loose when someone tries to link to this library because there can't be two main functions). We ultimately need a library. I'd avoid building an executable for now (we can build one later as a test, but it's going to add more complexity).